httpx/_transports/asgi.py

```diff
@@ -1,7 +1,5 @@
 import typing
 
-import sniffio
```

Comment:
`_utils.py` also imports sniffio as top-level.
Could we consider add httpcore's [`current_async_library`](https://github.com/encode/httpcore/blob/2fcd062df71555cc7de55774c6dc137551eb8692/httpcore/_synchronization.py#L21) in httpx?